feat: add C implementation for `stats/base/dists/exponential/quantile` (#4145)
stdlib-js/stdlib: lib/node_modules/@stdlib/stats/base/dists/exponential/quantile/README.md

<!-- /.examples -->

<!-- C interface documentation. -->

* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/stats/base/dists/exponential/quantile.h"
```

#### stdlib_base_dists_exponential_quantile( p, lambda )

> Returns a function for evaluating the [quantile function][quantile-function] of an [exponential][exponential-distribution] distribution with rate parameter `lambda`.

Reviewer: ```suggestion
Evaluates the [quantile function][quantile-function] for a [exponential][exponential-distribution] distribution with rate parameter `lambda`.
```

```c
double out = stdlib_base_dists_exponential_quantile( 0.8, 1.0 );
// returns ~1.609
```

The function accepts the following arguments:

> -   **p**: `[in] double` probability

Reviewer: ```suggestion
-   **p**: `[in] double` probability.
```

-   **lambda**: `[in] double` rate parameter.

```c
double stdlib_base_dists_exponential_quantile( const double p, const double lambda );
```

</section>

<!-- /.usage -->

<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

</section>

<!-- /.notes -->

<!-- C API usage examples. -->

<section class="examples">

### Examples

```c
#include "stdlib/stats/base/dists/exponential/quantile.h"
#include <stdlib.h>
#include <stdio.h>

static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v*(max-min) );
}

int main( void ) {
    double lambda;
    double y;
    double p;
    int i;

    for ( i = 0; i < 25; i++ ) {
        p = random_uniform( 0.0, 1.0 );
        lambda = random_uniform( 0.0, 20.0 );
        y = stdlib_base_dists_exponential_quantile( p, lambda );

>         printf( "p: %lf, λ: %lf, Q[p;λ]: %lf\n", p, lambda, y );

Reviewer: ```suggestion
        printf( "p: %lf, λ: %lf, Q(p;λ): %lf\n", p, lambda, y );
```
